Index: src/main/java/com/imcode/imcms/db/SingleConnectionDataSource.java
===================================================================
--- main/java/com/imcode/imcms/db/SingleConnectionDataSource.java	(revision 13193)
+++ main/java/com/imcode/imcms/db/SingleConnectionDataSource.java	(working copy)
@@ -8,7 +8,7 @@
 
 import com.imcode.db.jdbc.ConnectionWrapper;
 
-public class SingleConnectionDataSource implements DataSource {
+public abstract class SingleConnectionDataSource implements DataSource {
 
     private final Connection connection;
 
